remoting/src/main/java/org/apache/rocketmq/remoting/netty/NettyRemotingServer.java

```diff
@@ -169,6 +188,7 @@ public void initChannel(SocketChannel ch) throws Exception {
                             new NettyDecoder(),
                             new IdleStateHandler(0, 0, nettyServerConfig.getServerChannelMaxIdleTimeSeconds()),
                             new NettyConnectManageHandler(),
+                            new StartTlsHandler(),
```

Comment:
This `StartTls` mechanism is great, and there is another mechanism for your reference:

For SSL 3.0 or TLS 1.0, 1.1 and 1.2:

+-------+------------------+------------------+--------+------
| 0x16  | 2 bytes version  |  2 bytes length  |  0x01  |  etc.
+-------+------------------+------------------+--------+------

The first byte always is 0x16(SSL handshake), can be used to decide use SSL or not.